### PR TITLE
SimplifyConstCondition: Check moves for constants

### PR DESCRIPTION
**Pull request by clubby789**
Another possible take at #120650 - try and avoid codegenning our dead blocks

r? @ghost

### CODE REVIEW

File: compiler/rustc_mir_transform/src/simplify_branches.rs [hunk 1]
```diff
@@ -34,19 +34,45 @@ impl<'tcx> MirPass<'tcx> for SimplifyConstCondition {
...
+                        // produced by `if <constant>`
+                        if let Some(stmt) = block.statements.last()
+                            && let StatementKind::Assign(box (
+                                ref assigned,
+                                Rvalue::Use(Operand::Constant(ref c)),
+                            )) = stmt.kind
+                            && assigned == place
+                            && let Some(constant) = c.const_.try_eval_bits(tcx, param_env)
+                        {
+                            remove_last = true;
```
Comment (cjgillot) [inline compiler/rustc_mir_transform/src/simplify_branches.rs:58]: This is wrong if this is not the single use of _1.

**Comment by clubby789** [comment]
@rust-timer build 74b1060803ce097d501edc3ddedcf446b79d77c6

**Comment by clubby789** [comment]
Going to close this since #120650 went through

**Pull request closed without merging**